### ISSUE TITLE
Expand diagnostic quick fixes

### ISSUE THREAD

**Issue by lerebear** [astral-sh/ty#4138]
## Problem

To the extent possible, ty should offer an automated fix via the language server whenever it reports a diagnostic. ty already does offer missing-import suggestions, diagnostic suppressions, and selected diagnostic-provided fixes, but many useful corrections remain unavailable.

Comparable diagnostic fixes are documented for [Pylance](https://github.com/microsoft/pylance-release/blob/main/README.md#quick-fixes) and advertised by [Pyrefly](https://github.com/facebook/pyrefly/blob/b52e609470554983d33a2cb84074f619d5272d85/pyrefly/lib/lsp/non_wasm/server.rs#L1348-L1350).

For example, an incomplete enum match could offer to insert its missing cases:

```python
match color:
    case Color.RED:
        return "red"
```

## Candidate quick fixes

- [ ] Insert missing `match` arms.
- [ ] Add missing `@override` decorators.
- [ ] Generate a missing variable, function, or class from its usage.
- [ ] Generate type stubs for an unresolved or untyped dependency.
- [ ] Add inferred pytest fixture annotations.
- [ ] Assign an intentionally unused call result to `_`.
- [ ] Add or remove symbols from `__all__` and `__slots__`
- [ ] Expand existing diagnostic-specific fixes where a safe correction is available.

## Acceptance criteria

- [ ] Applicable diagnostics expose useful `quickfix` actions through the editor.
- [ ] Existing missing-import, suppression, and diagnostic-provided fixes continue to work.
- [ ] Unsafe, inapplicable, or ambiguous corrections are not offered as automatic fixes.
